src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -373,8 +373,13 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         val typeRef = impl.typeReference ?: return
         if (typeRef.traitType != null) return
         val type = typeRef.type
-        if (type is TyPrimitive && impl.queryAttributes.langAttribute == type.name) {
-            return // `#[lang = "u8"] impl u8 {}` is a valid impl
+        val langAttr = impl.queryAttributes.langAttribute
+        if ((type is TyPrimitive && langAttr == type.name) ||
+            (type is TyPointer && (langAttr == "const_ptr" || langAttr == "mut_ptr") ||
+                (type is TySlice && langAttr?.startsWith("slice") == true))) {
```

Comment:
Hm, this condition looks quite complicated
Maybe we can simplify it and check only existence of `lang` attribute? It's a bit less precise but greatly simplify logic and hardly work worse in the real world

Reply:
@Undin sure thing, will update later today